File: python-improperoutputhandling/vulnerable9.py
from jinja2 import Template
from openai import OpenAI

client = OpenAI()


def render_llm_response():
    response = client.chat.completions.create(
        model="gpt-4o-mini",
        messages=[{"role": "user", "content": "Write a short HTML snippet about: the history of the internet"}],
    )
    llm_html = response.choices[0].message.content
    template = Template("<div class='content'>{{ content }}</div>")
    return template.render(content=llm_html)


if __name__ == "__main__":
    print(render_llm_response())

File: javascript-systempromptleakage/vulnerable9.js
const { OpenAI } = require("openai");
const express = require("express");

const app = express();
const openai = new OpenAI();
const systemPrompt = "Internal tool. Access restricted. Contact admin@corp.internal for issues.";

app.use(express.json());

app.post("/chat", async (req, res) => {
    const completion = await openai.chat.completions.create({
        model: "gpt-4o",
        messages: [
            { role: "system", content: systemPrompt },
            { role: "user", content: req.body.message },
        ],
    });
    res.json({ reply: completion.choices[0].message.content });
});

app.use((err, req, res, next) => {
    res.status(500).json({
        error: err.message,
        context: systemPrompt,
    });
});

app.listen(3000);
